lib/rules/yo-version.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+var message = require('../message');
+var latestVersion = require('latest-version');
+var binVersionCheck = require('bin-version-check');
+
+exports.description = 'yo version';
+
+var errors = exports.errors = {
+  oldYoVersion: function () {
+    return message.get('yo-version-out-of-date', {});
+  }
+};
+
+exports.verify = function (cb) {
+  latestVersion('yo').catch(function (err) {
+    console.error(err);
+    cb(err);
+  }).then(function (version) {
+    if (typeof exports.OLDEST_YO_VERSION === 'undefined') {
```

Comment:
@SBoudrias I am not sure what you mean here.  You wan't me to remove the `if` statement, or you want me to not use `latest-version`, or other?